### DISCUSSION TITLE
Camera speaker as media player

### DISCUSSION THREAD

**Discussion by kongjudas** [JurajNyiri/HomeAssistant-Tapo-Control#675] [General]
I'm using BirdNet with several Tapo camera rtsp streams but is it possible to play recorded bird sounds thru the in-camera speaker? If not, is it even worth posting it as a feature request? I have also wanted to play a "You have been observed" kind of message that those speakers.

**Comment by blanknam3d** [comment]
2 weeks late, but: With go2rtc installed as an addon via either HACS or through Home Assistant's own addon store, add something like the following in your Home Assistant configuration.yaml:
```
media_player:
  - platform: webrtc
    entity_id: media_player.chirpo
    name: Birdwatching Camera
    stream: chirpo
    audio: auto
```

Then, add something like these lines to your go2rtc config:
```
streams:
  chirpo: tapo://InsertYourTapoCloudPwHere@192.168.x.xxx
```
Of course substitute "chirpo" for whatever you actually wanna call the camera (just has to be named the same in both go2rtc config and Home Assistant configuration.yaml), insert your actual cloud password, put in your camera's actual IP address, etc...

Once done, restart Home Assistant (or just reload your configs, though go2rtc will restart as you save the config), and you'll see them appear as media_player entities.

In my experience with the Tapo C200 cameras, I find that it often cuts off the audio a little bit at the start, and sometimes also cuts a little off the end too - so whatever audio you play, give it about 1200ms of silence at the start & end just to be sure it's not abruptly cut off.

**Reply by HybridZach** [reply]
Where / how to edit the go2rtc config? Its not in the config directory.

**Reply by blanknam3d** [reply]
> Where / how to edit the go2rtc config? Its not in the config directory.

Go to your go2rtc server's address at port 1984, ex. 192.168.0.XX:1984. Then go to config, and you will see an editor for the config.

It is also usually located at /homeassistant/go2rtc.yaml, if you would rather use the File editor addon to alter it - you will need to manually restart go2rtc afterward.

As for Home Assistant's configuration.yaml, that's in /homeassistant/configuration.yaml - [follow steps here for altering it](https://www.home-assistant.io/docs/configuration/)

(edit - further clarifying about Home Assistant's config yaml)

**Reply by HybridZach** [reply]
Hi thanks. Does this require a tapo cloud account? I am trying to get this setup for my C100 camera. Where the config requires entity id media_player.tapo I don't see that within home assist, but I know within the tapo app you can speak through the camera speaker. I want to use my model speaker is this possible?

**Reply by ghost** [reply]
I have cnfigured my tapo camera's via frigate, I'm now trying to get the speaker of the tapo as a media player in Home Assistant but can't figure out how. Help would be much appriciated!
